lib/Importer/ONNXModelLoader.cpp

```diff
@@ -2332,9 +2332,33 @@ Error ONNXModelLoader::loadMatMul(const ONNX_NAMESPACE::NodeProto &op,
   NodeValue RHS;
   ASSIGN_VALUE_OR_RETURN_ERR(RHS, getNodeValueByName(op.input(1)));
 
-  /// For dimension size equal to 3 use batchedMatMul
-  if (LHS.dims().size() == 3) {
+  /// For dimension greater than 2 use batchedMatMul
+  if (LHS.dims().size() > 2) {
+    dim_t numLHSDims = LHS.dims().size();
+    dim_t numRHSDims = RHS.dims().size();
+    std::vector<dim_t> finalShape;
+    for (auto d : LHS.dims()) {
+      finalShape.push_back(d);
+    }
```

Comment:
nit: Can we move this to the reshape branch after `createBatchMatMul`? It's not used inside the branch before `createBatchMatMul` except to set `finalShape`, but we could do that inside the reshape branch and keep it entirely in that if's scope.

Reply:
Done.